### DISCUSSION TITLE
After update ESP32 board package from V1.0.6 to V2.0.3, wifi can not connect anymore

### DISCUSSION THREAD

**Discussion by coolbechen** [espressif/arduino-esp32#8222] [General]
I am using ESP32 pico kit and ESP32-CAM for test under Arduino 1.8.13, if i use the following site to get the ESP32 board package V1.0.6 (or earlier), the example code (Advanced Web Server) or any other wifi related example codes all can connect to my office wifi AP in few seconds.
https://dl.espressif.com/dl/package_esp32_index.json 

But when i remove the V1.0.6 package, and change the site to 
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json
install V2.0.3 (or any others version), the same example code can not connect to my office wifi AP anymore. WiFi connection always fail.

Then I remove V2.0.3 and change back to the V1.0.6, the same example code can connect to my office wifi AP again.

Why? My office wifi AP is an old one, it it related? Because V1.0.6 do not support ESP32-S3 core, so i need to change the package to V2.0.9 for future use.

**Comment by coolbechen** [comment]
I found the problems. My office hub is an old one, using WEP for access control. If i install ESP32 sdk v1.0.6 and compile any wifi examples, all can connect to the AP in few seconds. But if i install ESP32 sdk 2.0.3 or later, the compiled any wifi examples all can not connect to the AP any more.

I change the office AP access control from WEP to WEP2, then the problems are solved.

There are still many this kind of old wifi hubs everywhere, so this problem is troubling me very much.

**Reply by me-no-dev** [reply]
WEP is almost equal to Open now. It's really insecure.

**Reply by mrengineer7777** [reply]
WEP was considered a security risk many years ago. I'm shocked anyone is still using it.

This repository also dropped support for WPA (not WPA2) a couple years ago. There's a function call to turn WPA back on.

**Reply by coolbechen** [reply]
Because WEP hub still exist in real world (not for public usage, in some factories or closed environment still use very old hub for internal link). My IOT devices need to use for it. What is the function call to turn WPA back on?

**Comment by mrengineer7777** [comment]
You can set the wifi minimum security with:

```
WiFi.setMinSecurity(WIFI_AUTH_WPA_PSK);                                           //Allow connecting to older WPA routers
WiFi.begin(netconf.wlan.ssid, netconf.wlan.password, chan, bssid , true);
```
Check the allowed parameter. There might be one for WEP. No guarantees the underlying IDF wifi driver will allow connecting to WEP.
